tests/kernel/sched/schedule_api/src/test_sched_priority.c

```diff
@@ -111,6 +111,19 @@ ZTEST(threads_scheduling, test_priority_preemptible)
 	k_thread_priority_set(k_current_get(), old_prio);
 }
 
+
+/* The acrn_ehl_crb board works in higher frequency, the
+ * time of starting a thread will be shorted, that might
+ * cause the lower priority thread t2 completed too early
+ * and the higher priority can not be scheduled in time.
+ * So let them have different scale of starting interval.
+ */
+#ifdef CONFIG_BOARD_ACRN
+#define START_INTERVAL 5
+#else
+#define START_INTERVAL 10
+#endif
+
```

Comment:
Maybe I am missing something, but I am not yet seeing how a higher board frequency or changing the thread should affect the test results.

I saw in your comments in the issue that in the failing log case, threads were created in the following order:
   T0, T1, T2, T3 (as expected)
And were awoken/entered in the following order:
   T0, T2, T1, T3 (as expected)

However, I do not yet understand how changing the delay times could impact the order in which the threads were removed from the *sync_sema* wait queue.